### PR TITLE
feat: add Analytics options to FlutterFire Starter

### PR DESCRIPTION
**Pull request by Lyokone**
## Description

Adds support for 2 analytics options in the FlutterFire starter template

## Type of Change

- [X] ✨ `feat` -- New feature (non-breaking change which adds functionality)
- [ ] 🛠️ `fix` -- Bug fix (non-breaking change which fixes an issue)
- [ ] ❌ `!` -- Breaking change (fix or feature that would cause existing functionality to change)
- [ ] 🧹 `refactor` -- Code refactor
- [ ] ✅ `ci` -- Build configuration change
- [ ] 📝 `docs` -- Documentation
- [ ] 🗑️ `chore` -- Chore

### CODE REVIEW

File: packages/flutterfire_starter/__brick__/lib/src/sample_feature/sample_item_details_view.dart [hunk 1]
```diff
@@ -2,15 +2,16 @@ import 'package:flutter/material.dart';
 
 /// Displays detailed information about a SampleItem.
 class SampleItemDetailsView extends StatelessWidget {
-  const SampleItemDetailsView({Key? key}) : super(key: key);
+  const SampleItemDetailsView({Key? key, this.itemId}) : super(key: key);
 
-  static const routeName = '/sample_item';
+  static const routeName = '{{^analyticswithgorouter}}/{{/analyticswithgorouter}}sample_item';
```
Comment (russellwheatley) [inline packages/flutterfire_starter/__brick__/lib/src/sample_feature/sample_item_details_view.dart:7]: Just checking if there should be a forward slash before `sample_item`? `{{/analyticswithgorouter}}/sample_item`

File: packages/flutterfire_starter/hooks/post_gen.dart [hunk 2]
```diff
@@ -34,6 +38,24 @@ Future<void> _installDependencies(HookContext context) async {
...
+  final varsPlugins = (context.vars['plugins'] as List<dynamic>).cast<String>();
+  final varsPluginsName = varsPlugins.map((e) => e.split(' ')[0]).toList();
+  if (varsPluginsName.contains('Analytics')) {
+    await Process.run(
+      'flutter',
+      ['pub', 'add', 'firebase_analytics'],
+      workingDirectory: './$appName',
+    );
+  }
+  if (varsPlugins.contains('Analytics with GoRouter')) {
```
Comment (russellwheatley) [inline packages/flutterfire_starter/hooks/post_gen.dart:51]: possibly make a note that this is the value pulled from the brick.yaml (I think) so it's understood where it is coming from.

File: packages/flutterfire_starter/hooks/pre_gen.dart [hunk 3]
```diff
@@ -12,12 +15,40 @@ Future<void> run(HookContext context) async {
...
-Future<ProcessResult> _generateApp(HookContext context) async {
+void _validatePluginList(HookContext context) {
+  final varsPlugins =
+      ((context.vars['plugins'] as List<dynamic>).cast<String>())
+          .map((e) => e.split(' ')[0])
+          .toList();
+  final setPlugins = varsPlugins.toSet().toList();
+  if (setPlugins.length != varsPlugins.length) {
+    context.logger.err(
+      'You have duplicate plugins in your plugins list. Please remove them.',
```
Comment (russellwheatley) [inline packages/flutterfire_starter/hooks/pre_gen.dart:26]:
```suggestion
      'You have duplicate plugins in your plugin's list. Please remove them.',
```

File: packages/flutterfire_starter/hooks/pre_gen.dart [hunk 4]
```diff
@@ -12,12 +15,40 @@ Future<void> run(HookContext context) async {
...
+  }
+}
+
+Future<void> _generateApp(HookContext context) async {
   context.logger.info('Running flutter create...');
   final appName = context.vars['name'] as String;
   final appDescription = context.vars['description'] as String;
   final nameOrg = context.vars['org'] as String;
-  return Process.run('flutter', [
+  await Process.run('flutter', [
```
Comment (russellwheatley) [inline packages/flutterfire_starter/hooks/pre_gen.dart:51]: Not sure the exception would bubble up from this process. Might be worth adding something like:

```dart
final runScript = await Process.run('flutter', [
    'create',
    appName,
    '-t',
    'skeleton',
    '--description',
    appDescription,
    '--org',
    nameOrg,
  ]);

if (runScript.exitCode != 0) {
   throw Exception(runScript.stderr);
 }
````

**Review by russellwheatley — CHANGES_REQUESTED** [review]

File: packages/flutterfire_starter/__brick__/lib/src/sample_feature/sample_item_details_view.dart (hunk 1 above)
Comment (Lyokone) [reply packages/flutterfire_starter/__brick__/lib/src/sample_feature/sample_item_details_view.dart:7]: When using GoRouter you have to remove the forward slash, so it's okay this way

File: packages/flutterfire_starter/hooks/post_gen.dart (hunk 2 above)
Comment (Lyokone) [reply packages/flutterfire_starter/hooks/post_gen.dart:51]: I've added that

**Review by russellwheatley — APPROVED** [review]

**Pull request merged**